### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by NyraO**

### CODE REVIEW

File: assessment_software/src/app/app.component.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <p>Hello World</p>
+<router-outlet></router-outlet>
```
Comment (Aben-Bel) [inline assessment_software/src/app/app.component.html:2]: This should be the only `<router-outlet></router-outlet>` in the whole angular project for our case. This router-outlet will be substituted with the matching page according to the URL. 

Remove other `<router-outlet></router-outlet>` from other pages.

File: assessment_software/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,7 +1,36 @@
...
+  { path: 'change_password', component: ChangePasswordComponent },
+  { path: 'courses', component: HomeComponent, 
+      children:[
+        { path: 'course_name', component: CourseComponent, 
+            children:[
+            { path: 'edit assignment', component: EditAssignmentComponent  },
+            { path: 'add assignment', component: AddAssignmentComponent },
+            { path: 'student_list', component: StudentListComponent },
+            {path: 'assignment', component: AssignmentComponent,
+                children:[{path: 'assignment_name', component: AssignmentDetailComponent}]}
```
[the remark this answers is not in the document]
Comment (Aben-Bel) [reply assessment_software/src/app/app-routing.module.ts:29]: For some reason I don't know, the current nesting with children is not working when I tested. It only works when you do it this way:

```
  { path: 'courses', component: HomeComponent },
  { path: 'courses/:courseId', component: CourseComponent },
  {
    path: 'courses/:courseId/edit_assignment',
    component: EditAssignmentComponent,
  },
  {
    path: 'courses/:courseId/student_list',
    component: StudentListComponent,
  },
  {
    path: 'courses/:courseId/add_assignment',
    component: AddAssignmentComponent,
  },
  {
    path: 'courses/:courseId/assignment/:assignmentId',
    component: AssignmentDetailComponent,
  },
```

Change this line of code to the above one:

```
  {
    path: 'courses',
    component: HomeComponent,
    children: [
      {
        path: ':course_name',
        component: CourseComponent,
        children: [
          { path: 'edit assignment', component: EditAssignmentComponent },
          { path: 'add assignment', component: AddAssignmentComponent },
          { path: 'student_list', component: StudentListComponent },
          {
            path: 'assignment',
            component: AssignmentComponent,
            children: [
              {
                path: ':assignment_name',
                component: AssignmentDetailComponent,
              },
            ],
          },
        ],
      },
    ],
  },
```

File: assessment_software/src/app/page/add assignment/add-assignment/add-assignment.component.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<p>add-assignment works!</p>
+<router-outlet></router-outlet>
```
Comment (Aben-Bel) [inline assessment_software/src/app/page/add assignment/add-assignment/add-assignment.component.html:2]: remove <router-outlet></router-outlet>

File: assessment_software/src/app/page/assignment/assignment.component.html [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<p>assignment works!</p>
+<router-outlet></router-outlet>
```
Comment (Aben-Bel) [inline assessment_software/src/app/page/assignment/assignment.component.html:2]: remove router-outlet

File: assessment_software/src/app/page/course/course/course.component.html [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<p>course works!</p>
+<router-outlet></router-outlet>
```
Comment (Aben-Bel) [inline assessment_software/src/app/page/course/course/course.component.html:2]: remove router-outlet

File: assessment_software/src/app/page/home/home/home.component.html [hunk 6]
```diff
@@ -0,0 +1,2 @@
+<p>home works!</p>
+<router-outlet></router-outlet>
```
Comment (Aben-Bel) [inline assessment_software/src/app/page/home/home/home.component.html:2]: remove router-outlet

File: assessment_software/src/app/app-routing.module.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (NyraO) [reply assessment_software/src/app/app-routing.module.ts:29]: it works without any changes
image

**Review by Aben-Bel — APPROVED** [review]
Great Job! On fire these days 🔥

**Pull request merged**